packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -204,9 +225,89 @@ const reducer: Reducer<State> = (state = INITIAL_STATE, action) => {
   return state;
 };
 
+const confirmWriteOperationIfNeeded = async (
+  instance: MongoDBInstance,
+  dataService: DataService,
+  namespace: string,
+  pipeline: Document[]
+) => {
+  if (!pipeline.length) {
+    return true;
+  }
+
+  const lastStageOperator = getStageOperator(pipeline[pipeline.length - 1]);
+  let typeOfWrite;
+
+  if (!lastStageOperator || !isOutputStage(lastStageOperator)) {
+    return true;
+  }
+
+  const destinationNamespace = getDestinationNamespaceFromStage(namespace, {
+    [lastStageOperator]: pipeline[pipeline.length - 1][lastStageOperator],
+  });
```

Comment:
This is just this, right?

```suggestion
  const destinationNamespace = getDestinationNamespaceFromStage(namespace, pipeline[pipeline.length - 1]);
```

Reply:
Lol yeah 🙈